src/test/scala/ai/privado/languageEngine/go/passes/orm/GormParserTest.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+package ai.privado.languageEngine.go.passes.orm
+
+import ai.privado.languageEngine.go.tagger.GoTaggingTestBase
+import ai.privado.model.*
+import ai.privado.semantic.Language.*
+import io.joern.x2cpg.X2Cpg
+import io.shiftleft.semanticcpg.language.*
+import io.shiftleft.semanticcpg.layers.LayerCreatorContext
+import io.joern.dataflowengineoss.layers.dataflows.{OssDataFlow, OssDataFlowOptions}
+
+class GormParserTest extends GoTaggingTestBase {
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    X2Cpg.applyDefaultOverlays(cpg)
+
+    val context = new LayerCreatorContext(cpg)
+    val options = new OssDataFlowOptions()
+    new OssDataFlow(options).run(context)
+    new GormParser(cpg).createAndApply()
+  }
+
+  override val goFileContents =
```

Comment:
In this test Case, which method gets matched with `cpg.method.where(_.parameter.typeFullName(GORM_PARAMETER_TYPE_RULE)).typeDecl.dedup.toArray` this query. ?
As I understand the Ty peDecl node we get will be of `User`

Reply:
`Save` method will get matched to query. And `Save` is part of TypeDecl `User`.

Reply:
Ok, looks like this is handled as part of AST creation differently. 

My only concern is False positives due to this, I hope you have already tested this across different cases, if YES that should be good

Reply:
Just to simplify purpose of query "Get all TypeDecl node which are dealing with GORM dependency".
TypeDecl node are only created when they actually defined in code base. As parse is tightly coupled with GORM I don't see any case where false positive may arise.